### PR TITLE
perf: generate dep optimizations

### PR DESCRIPTION
**Pull request by kevjue**

### CODE REVIEW

File: core/src/bytes/event.rs [hunk 1]
```diff
@@ -9,28 +11,41 @@ use serde::{Deserialize, Serialize};
...
     /// The first input operand.
-    pub b: u32,
+    pub b: u8,
 
     /// The second input operand.
-    pub c: u32,
+    pub c: u8,
+}
+
+impl Hash for ByteLookupEvent {
```
Comment (jtguibas) [inline core/src/bytes/event.rs:38]: i'm skeptcail this is faster

Comment (jtguibas) [reply core/src/bytes/event.rs:38]: did you try it?

Comment (kevjue) [reply core/src/bytes/event.rs:38]: Ya, I did.  It did speed it up significantly.  

The original implementation will hash each individual elements, e.g.

```
impl $crate::hash::Hash for ByteLookupEvent {
    fn hash<H: $crate::hash::Hasher>(&self, ra_expand_state: &mut H) {
        match self {
            ByteLookupEvent {
                shard: shard,
                channel: channel,
                opcode: opcode,
                a1: a1,
                a2: a2,
                b: b,
                c: c,
            } => {
                shard.hash(ra_expand_state);
                channel.hash(ra_expand_state);
                opcode.hash(ra_expand_state);
                a1.hash(ra_expand_state);
                a2.hash(ra_expand_state);
                b.hash(ra_expand_state);
                c.hash(ra_expand_state);
            }
        }
    }
}
```

So both the number of hash invocations and the amount of data it's hashing is improved.

Comment (kevjue) [reply core/src/bytes/event.rs:38]: It improved the gen dep time by ~25% for the chips with lots of these events (e.g. CPU, AddSub).

**Review by jtguibas — APPROVED** [review]

**Pull request merged**